src/app/components/SendForm/ChainSend.tsx

```diff
@@ -0,0 +1,272 @@
+import { isPossibleDust } from 'utils/validators';
+import './ChainSend.less';
+
+interface StateProps {
+  account: AppState['account']['account'];
+  chain: ReturnType<typeof getNodeChain>;
+  sendOnChainReceipt: AppState['payment']['sendOnChainReceipt'];
+  isSending: AppState['payment']['isSending'];
+  sendError: AppState['payment']['sendError'];
+  onChainFees: AppState['payment']['onChainFees'];
+  feesError: AppState['payment']['feesError'];
+  isFetchingFees: AppState['payment']['isFetchingFees'];
+}
+
+interface DispatchProps {
+  sendOnChain: typeof sendOnChain;
+  resetSendPayment: typeof resetSendPayment;
+  getOnChainFeeEstimates: typeof getOnChainFeeEstimates;
+}
+
+interface OwnProps {
+  close?(): void;
+}
+
+type Props = StateProps & DispatchProps & OwnProps;
+
+interface State {
+  amount: string;
+  isSendAll: boolean;
+  address: string;
+  feeMethod: string;
+  fee: number;
+  showMoreInfo: boolean;
+}
+
+const INITIAL_STATE = {
+  amount: '',
+  isSendAll: false,
+  address: '',
+  feeMethod: '',
+  fee: 0,
+  showMoreInfo: false,
+};
+
+class ChainSend extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      ...INITIAL_STATE
+    }
+  }
+
+  componentWillReceiveProps(nextProps: Props) {
+    const { onChainFees } = this.props;
+    if (nextProps.onChainFees !== onChainFees && nextProps.onChainFees !== null) {
+      this.setState({ 
+        feeMethod: 'fastestFee',
+        fee: nextProps.onChainFees.fastestFee,
+        showMoreInfo: true,
+      });
+    }
+  }
+
+  render() {
+    const { 
+      sendOnChainReceipt, 
+      isSending, 
+      sendError, 
+      account, 
+      onChainFees, 
+      isFetchingFees, 
+      feesError 
+    } = this.props;
+    
+    // Early exit for send state
+    if (isSending || sendOnChainReceipt || sendError) {
+      return (
+        <SendState
+          isLoading={isSending}
+          error={sendError}
+          result={sendOnChainReceipt && (
+            <>
+              <h3>Transaction ID</h3>
+              <a
+                href={`https://blockstream.info/tx/${sendOnChainReceipt.txid}`}
+                target="_blank"
+                rel="noopener nofollow"
+              >
+                {sendOnChainReceipt.txid}
+              </a>
+            </>
+          )}
+          back={this.props.resetSendPayment}
+          reset={this.reset}
+          close={this.props.close}
+        />
+      );
+    }
+
+    const { amount, isSendAll, address, fee, feeMethod, showMoreInfo } = this.state;
+    const blockchainBalance = account ? account.blockchainBalance : '';
+    const disabled = (!amount && !isSendAll) || !address ||
+      (!!blockchainBalance && !!amount && (new BN(blockchainBalance).lt(new BN(amount))));
+    const dustWarning = isPossibleDust(amount, address, fee) ? (
+      <><Icon type="exclamation-circle" /> Dust warning: This amount may not be spendable</>
+    ) : undefined;
```

Comment:
I could never actually get this to show up, no matter how small I made the remainder. Not sure if there's something I was doing wrong here though.

Reply:
Bah, I think I know. The fee is required to calculate the dust threshold but the fee doesn’t get fetched until the Advanced button is clicked. This will be fixed once I make the fee always visable.